File: trail2/블럭쌓는 명령2/block-stacking-commands2.py
n, k = map(int, input().split())
commands = [tuple(map(int, input().split())) for _ in range(k)]

arr = [0] * n;

for a, b in commands:
    for i in range(a - 1, b):
        arr[i] += 1


max = arr[0]
for i in range(1,n):
    if (max < arr[i]):
        max = arr[i]
    else:
        continue

print(max)

File: trail2/블럭쌓는 명령2/block-stacking-commands2.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int K = sc.nextInt();

        int[] arr = new int[N];

        for (int i = 0; i < K; i++) {
            int A = sc.nextInt();
            int B = sc.nextInt();

            for (int j = A - 1; j < B; j++) {
                arr[j] += 1;
            }
        }

        int max = arr[0];
        for (int i = 1; i < N; i++) {
            if (max < arr[i]) {
                max = arr[i];
            }
            else {
                continue;
            }
        }
        
        System.out.print(max);
    }
}